BinaryEncoding.md

```diff
@@ -65,20 +65,40 @@ Yes:
 
 ## Global structure
 
-* A module contains:
-  * a header followed by
-  * a table (sorted by offset) containing, for each section, its type and offset (within the module), followed by
-  * a sequence of sections.
+* A module contains (in this order):
+  - A header, containing:
+    + The [magic number](https://en.wikipedia.org/wiki/Magic_number_%28programming%29)
+    + Other data TBD (possibly entrypoint, memory bitness, source language, etc.)
```

Comment:
In the AstSemantics.md, we've just been assuming that one can index the
linear memory with either 32-bit or 64-bit offsets. In the v8 native
prototype, there are different bytecode numbers for whether the memory
offset operand is an Int32 or an Int64.


On Tue, Jun 30, 2015 at 10:29 PM, Michael Holman <notifications@github.com>
wrote:

> In BinaryEncoding.md
> <https://github.com/WebAssembly/design/pull/237#discussion_r33619752>:
>
> > @@ -65,20 +65,40 @@ Yes:
> >
> >  ## Global structure
> >
> > -* A module contains:
> > -  * a header followed by
> > -  * a table (sorted by offset) containing, for each section, its type and offset (within the module), followed by
> > -  * a sequence of sections.
> > +* A module contains (in this order):
> > +  - A header, containing:
> > +    + The [magic number](https://en.wikipedia.org/wiki/Magic_number_%28programming%29)
> > +    + Other data TBD (possibly entrypoint, memory bitness, source language, etc.)
>
>  What is memory bitness?
>
> I mean whether your linear memory has a 64 bit or 32 bit address space
> (i.e. whether ptr type is int32 or int64). Maybe not necessary here, but
> was just an idea for something we might want. I think we might need some
> version info for the module format as well. We may never need to break
> compat, but I can imagine a scenario where we eventually want to make
> format changes, and having a byte to allow for that would be useful.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebAssembly/design/pull/237/files#r33619752>.
>